modules/transport-netty4/src/test/java/org/elasticsearch/rest/discovery/Zen2RestApiIT.java

```diff
@@ -62,13 +64,27 @@ protected Settings nodeSettings(int nodeOrdinal) {
             .put(GatewayService.RECOVER_AFTER_MASTER_NODES_SETTING.getKey(), 1)
             .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), Integer.MAX_VALUE);
 
-        if (nodeOrdinal == 0) {
-            builder.put(ClusterBootstrapService.INITIAL_MASTER_NODE_COUNT_SETTING.getKey(), 2);
-        }
-
         return builder.build();
     }
 
+    @Override
+    protected List<Settings> updateNodesSettings(List<Settings> allNodesSettings) {
+        final Settings firstNodeSettings = allNodesSettings.get(0);
+        final List<Settings> otherNodesSettings = allNodesSettings.subList(1, allNodesSettings.size());
+        final List<String> masterNodeNames = allNodesSettings.stream()
+                .filter(s -> org.elasticsearch.node.Node.NODE_MASTER_SETTING.get(s))
+                .map(s -> org.elasticsearch.node.Node.NODE_NAME_SETTING.get(s))
```

Comment:
Could just be

```
        final List<String> masterNodeNames 
            = allNodesSettings.stream().filter(NODE_MASTER_SETTING::get).map(NODE_NAME_SETTING::get).collect(Collectors.toList());
```

Reply:
Sure, I've fixed 3 occurrences by this commit https://github.com/elastic/elasticsearch/pull/36559/commits/65ba82dbf3ed8380d0366c9ce1689aae5d93e26e